src/main/java/seedu/address/logic/commands/PersonAddSubGroupCommand.java

```diff
@@ -62,13 +62,11 @@ public CommandResult execute(Model model) throws CommandException {
             superGroup = new SuperGroup(groupName);
             model.addSuperGroup(superGroup);
         }
-
-        SubGroup subGroup = model.findSubGroup(groupName + "_" + subGroupName);
+        SubGroup subGroup = superGroup.findSubGroup(subGroupName);
         if (subGroup == null) {
             subGroup = new SubGroup(subGroupName, superGroup.getName());
```

Comment:
Could be a method i.e. superGroup.addSubGroup(subGroupName) just asks the supergroup to call the constructor instead and add it to the list. Better use of SLAP imo! :)